### ISSUE TITLE
Bug with smart quotes in a string

### ISSUE THREAD

**Issue by tverweij** [dotnet/vblang#503]
@KathleenDollard , @AnthonyDGreen 
We just found a bug in VB.Net.

VB supports smart quotes around a string, so the following is a valid statement
Dim x = "bla bla “” bla bal" 'All smart quotes - this compiles.

But it returns:
bla bla " bla bal 
The quote returned is a NORMAL - NON SMART quote.

This causes a problem where smart quotes can not be used in strings.

A proper way to handle this is that it would add the second quote to the string, as the first one is the escaper of the used quote.

**Comment by Happypig375** [comment]
Not really sure if this is by design or not.

**Comment by Padanian** [comment]
All bugs come by design.

**Comment by tverweij** [comment]
@Happypig375: show me the spec that states this.

**Comment by KathleenDollard** [comment]
The VB language is the spec. If something has been a particular way for a long time, we would think carefully about changing it for backwards compatibility. 

On this: 
Is this a recent change in compiler behavior? No
Is there a workaround? While painful, you can enter the code
Is it possible for this to break existing applications? Yes

So, my first reaction is that we would not take it. But that is why, and I don't mind pushback if I'm wrong on that.

**Comment by tverweij** [comment]
> The VB language is the spec. 
Wow, that means there *can not* be bugs in it :-)

We found this one and I thought - lets report it.

**Comment by VBAndCs** [comment]
VB compiler tests use a clever trick to avoid issues of embedded quotes, lines and other special symbols in the string:
```VB.NET 
Dim x = <![CDATA[bla bla “” bla bal]]>.Value
```

It returns: `bla bla “” bla bal`
I wish is this can be simplifies like this:
```VB.NET 
Dim x = $bla bla “” bla bal$
```

Every thing whatever between `$ $` is taken as it is, including lines and single $. If some one needs to embed `$` he wscape it by using extra `$`:

```VB.NET 
Dim x = $bla bla “” 5$$ bla bal$
```

**Comment by RevensofT** [comment]
Hi @tverweij ,
Isn't it always double characters in string is escape character in VB string ? 
```vb
Dim Esc_quote = "he reply ""I don't know"" to my question."
Dim Esc_block = $"When I ask him {{something}}, {Esc_quote}"
```

**Comment by tverweij** [comment]
@RevensofT Yes, to use a double quote in a string, you have to escape it with another one.

**Comment by RevensofT** [comment]
I think they replace it with double those character instead, personally I think double character escape is much more easier to read and write then \ / "" escape.

**Comment by WolvenRA** [comment]
@tverweij 
>  
> VB supports smart quotes around a string, so the following is a valid statement
> Dim x = "bla bla “” bla bal"    'All smart quotes - this compiles.
> 
> But it returns:
> bla bla " bla bal

Maybe I don't understand what you're trying to say about "smart quotes", but in my experience that isn't an error.  That's what it's supposed to do with the 2 double quotes.  Simply adding a space between the two double quotes would solve the problem if you don't want it there.  i.e.

Dim x = "bla bla “ ” bla bal"    Would return bla bla  bla bal    If that's what your looking for.

**Comment by tverweij** [comment]
@WolvenRA Note that none of the two double quotes in the string is chr(34)
And the code returned is.
